packages/next/src/server/request-adapter/standalone-request-adapter.ts

```diff
@@ -0,0 +1,32 @@
+import type { BaseNextRequest } from '../base-http'
+import type { NextUrlWithParsedQuery } from '../request-meta'
+import type { RequestAdapter } from './request-adapter'
+import type { XInvokePathRequestAdapter } from './x-invoke-path-request-adapter'
+import type { XMatchedPathRequestAdapter } from './x-matched-path-request-adapter'
+
+export class StandaloneRequestAdapter<ServerRequest extends BaseNextRequest>
+  implements RequestAdapter<ServerRequest>
+{
+  constructor(
+    private readonly xMatchedPathRequestAdapter: XMatchedPathRequestAdapter<ServerRequest>,
+    private readonly xInvokePathRequestAdapter: XInvokePathRequestAdapter<ServerRequest>
+  ) {}
+
+  public async adapt(
+    req: ServerRequest,
+    parsedURL: NextUrlWithParsedQuery
+  ): Promise<void> {
+    // Today, standalone mode is used to test the x-matched-path support as
+    // well.
+
+    // FIXME: remove this fallback when the tests are updated.
```

Comment:
Since this adapter isn't actually doing anything and is just forwarding it to an adapter that does, I feel like it makes more sense to branch between the relevant adapters at the call-site rather than making a new adapter that does it (since it seems like a "standalone" adapter should never exist -- it's either x-matched-path or x-invoke-path)

It didn't immediately make sense to me why the adapter consumed two other adapters and would probably be a bit clearer if we changed that

Reply:
This is temporary, as the only reason that the standalone version also uses `x-matched-path` is so we can use it for testing the implementation as it would run when deployed (see the [required-server-files](https://github.com/vercel/next.js/blob/c01ee2373b83860899bb5a33fad304f54ae11c0d/test/production/standalone-mode/required-server-files) directory of tests). Once the tests have either been moved out of there and into standalone unit tests, or we have a more compact way of building for deploying to Vercel, then we can remove this special handling. I think it keeps the weirdness isolated until we can update the tests.